# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: ipc/chromium/src/chrome/common/child_thread.cc
# Commit: 115187a5f512
# Full Hash: 115187a5f5120c5fda63685fd8a1471339852735
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-06-26 09:46:54
# Description:
#   Bug 1717993 - Hold a WeakPtr to mExistingListener in NodeChannel, r=handyman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D118868
# ==============================================================================

diff -r d0b8984194c6 -r 115187a5f512 ipc/chromium/src/chrome/common/child_thread.cc
--- a/ipc/chromium/src/chrome/common/child_thread.cc	Fri Jun 25 21:10:50 2021 +0000
+++ b/ipc/chromium/src/chrome/common/child_thread.cc	Fri Jun 25 21:30:49 2021 +0000
@@ -17,27 +17,20 @@
   channel_name_ = IPC::Channel::ChannelIDForCurrentProcess();
 }
 
-ChildThread::~ChildThread() {}
+ChildThread::~ChildThread() = default;
 
 bool ChildThread::Run() {
   bool r = StartWithOptions(options_);
   return r;
 }
 
-void ChildThread::OnChannelError() {
-  RefPtr<mozilla::Runnable> task = new MessageLoop::QuitTask();
-  owner_loop_->PostTask(task.forget());
-}
-
-void ChildThread::OnMessageReceived(IPC::Message&& msg) {}
-
 ChildThread* ChildThread::current() {
   return ChildProcess::current()->child_thread();
 }
 
 void ChildThread::Init() {
   auto channel = mozilla::MakeUnique<IPC::Channel>(
-      channel_name_, IPC::Channel::MODE_CLIENT, this);
+      channel_name_, IPC::Channel::MODE_CLIENT, nullptr);
 
   initial_port_ =
       mozilla::ipc::NodeController::InitChildProcess(std::move(channel));